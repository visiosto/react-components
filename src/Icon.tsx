// Copyright (c) 2021 Visiosto oy
// Licensed under the MIT License

import React from 'react';
import styled, { css } from 'styled-components';

const Span = styled.span<{ size: number }>`
  ${(props) => {
    if (props.size === 18) {
      return css`
        margin: 0;
        vertical-align: -5px;
      `;
    }

    if (props.size === 24) {
      return css`
        margin: 0;
        vertical-align: -6px;
      `;
    }

    return null;
  }}
`;

export type IconProps = {
  className?: string | undefined;
  icon:
    | 'arrow_back'
    | 'arrow_forward'
    | 'auto_awesome'
    | 'cancel'
    | 'check'
    | 'done'
    | 'event'
    | 'navigate_before'
    | 'navigate_next'
    | 'open_in_new'
    | 'public'
    | 'search'
    | 'send'
    | 'tune';
  size?: 18 | 24 | 36 | 48 | 64;
};

function Icon({ className = undefined, icon, size = 24 }: IconProps) {
  if (className) {
    return (
      <Span className={`material-icons md-${size} ${className}`} size={size}>
        {icon}
      </Span>
    );
  }

  return (
    <Span className={`material-icons md-${size}`} size={size}>
      {icon}
    </Span>
  );
}

export default Icon;
