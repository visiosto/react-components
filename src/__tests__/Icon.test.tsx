// Copyright (c) 2021 Visiosto oy
// Licensed under the Apache License, Version 2.0

import React from 'react';

import Icon from '../Icon';

import renderWithProviders from '../util/test/renderWithProviders';

describe('Icon component', () => {
  it('renders correctly', () => {
    const { container } = renderWithProviders(<Icon icon="tune" />);

    expect(container).toMatchSnapshot();
  });

  it('renders outlined icons correctly', () => {
    const { container } = renderWithProviders(<Icon icon="cloud" iconStyle="outlined" />);

    expect(container).toMatchSnapshot();
  });

  it('renders filled icons correctly', () => {
    const { container } = renderWithProviders(<Icon icon="cloud" iconStyle="filled" />);

    expect(container).toMatchSnapshot();
  });

  it('renders rounded icons correctly', () => {
    const { container } = renderWithProviders(<Icon icon="cloud" iconStyle="round" />);

    expect(container).toMatchSnapshot();
  });

  it('renders sharp icons correctly', () => {
    const { container } = renderWithProviders(<Icon icon="cloud" iconStyle="sharp" />);

    expect(container).toMatchSnapshot();
  });

  it('renders two-tone icons correctly', () => {
    const { container } = renderWithProviders(<Icon icon="cloud" iconStyle="two-tone" />);

    expect(container).toMatchSnapshot();
  });
});
