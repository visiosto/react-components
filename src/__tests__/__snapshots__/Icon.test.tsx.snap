// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders correctly 1`] = `
<div>
  <span
    class="Icon__Span-sc-1brds2j-0 gHKTxp material-icons md-24"
    size="24"
  >
    tune
  </span>
</div>
`;
